gdx/src/com/badlogic/gdx/graphics/g3d/particles/batches/PointSpriteParticleBatch.java

```diff
@@ -96,34 +113,78 @@ public PointSpriteParticleBatch (int capacity, ParticleShader.Config shaderConfi
 			this.blendingAttribute = new BlendingAttribute(GL20.GL_ONE, GL20.GL_ONE_MINUS_SRC_ALPHA, 1f);
 		if (this.depthTestAttribute == null) this.depthTestAttribute = new DepthTestAttribute(GL20.GL_LEQUAL, false);
 
-		allocRenderable();
+		renderables = new Array<Renderable>();
+		renderablePool = new RenderablePool();
+		blendingAttribute = new BlendingAttribute(GL20.GL_ONE, GL20.GL_ONE_MINUS_SRC_ALPHA, 1f);
+		depthTestAttribute = new DepthTestAttribute(GL20.GL_LEQUAL, false);
```

Comment:
I am not quite sure I understand this part. A few lines above we assign `this.blendingAttribute` and `this.depthTestAttribute` from the parameters. If this assignment brought `null` then we create new attributes there.
And at this point, we override the parameters with yet another new objects.
Is it needed? If we need to pass the attributes as out parameters, does it make sense here to assign recently created `this.blendingAttribute` and `this.depthTestAttribute`, not making new ones ?

Reply:
Oh, i missed that during the rebase. Thank you very much!